bridges/snowbridge/pallets/system-v2/src/lib.rs

```diff
@@ -79,20 +78,19 @@ pub mod pallet {
 		/// Send messages to Ethereum
 		type OutboundQueue: SendMessage;
 
-		/// Origin check for XCM locations that can create agents
+		/// Origin check for XCM locations that transact with this pallet
 		type SiblingOrigin: EnsureOrigin<Self::RuntimeOrigin, Success = Location>;
```

Comment:
```suggestion
		/// Origin check for locations acting as a frontend for this pallet
		type FrontendOrigin: EnsureOrigin<Self::RuntimeOrigin, Success = Location>;
```

I added another comment related to this. You should rename this to `FrontendOrigin` and make sure the implementation of `FrontendOrigin` checks that the location is like `/Para(AssetHub)/PalletIndex(SystemFrontend)`.

Reply:
Just curious is this nessessary?  Currently `SiblingOrigin`  is configured to accept only the root location of AH(1, Parachain(1000)) which I'd assume can't be manipulated by user origin.

Reply:
We do this because its good practice to be specific and restrictive when authorizing calls. The root origin `AH(1, Parachain(1000))` doesn't give us any indication of what pallet on AH is making the call.

In any case, this is what we in the inbound-queue for BH->AH xcms. Parity asked us to do it. And they are going to hit us with same feedback unless we do it now.

Reply:
https://github.com/vgeddes/polkadot-sdk/pull/6/commits/15f96644b1318f28927943aedfdc1280aeb8af5d